src/controllers/update/update.filing.date.controller.ts

```diff
@@ -1,22 +1,42 @@
 import { NextFunction, Request, Response } from "express";
 import { Session } from "@companieshouse/node-session-handler";
 
-import { logger } from "../../utils/logger";
+import { createAndLogErrorRequest, logger } from "../../utils/logger";
 import * as config from "../../config";
 import { isActiveFeature } from "../../utils/feature.flag";
-import { getApplicationData, setExtraData, mapFieldsToDataObject, mapDataObjectToFields } from "../../utils/application.data";
+import { getApplicationData, setExtraData, mapFieldsToDataObject, mapDataObjectToFields, addPrivateOverseasEntityDetailsToAppData } from "../../utils/application.data";
 import { postTransaction } from "../../service/transaction.service";
 import { createOverseasEntity, updateOverseasEntity } from "../../service/overseas.entities.service";
-import { OverseasEntityKey, Transactionkey, InputDateKeys } from '../../model/data.types.model';
+import { OverseasEntityKey, Transactionkey, InputDateKeys, EntityNumberKey } from '../../model/data.types.model';
 import { FilingDateKey, FilingDateKeys } from '../../model/date.model';
 import { ApplicationData } from "../../model/application.model";
+import { getPrivateOeDetails } from "../../service/private.overseas.entity.details";
 
-export const get = (req: Request, res: Response, next: NextFunction) => {
+const fetchAndSavePrivateOeData = async (req: Request, appData: ApplicationData): Promise<void> => {
+  const companyNumber = appData[EntityNumberKey];
+
+  if (!companyNumber) {
+    throw createAndLogErrorRequest(req, "Unable to request private OE data. Missing entity number in application data");
+  }
```

Comment:
this route has a navcheck for `hasOverseasEntity` so I don't think this code of !companyNumber will ever be a genuine scenario and if it's needed or not.